src/pubkey.cpp

```diff
@@ -188,6 +188,17 @@ bool XOnlyPubKey::CheckPayToContract(const XOnlyPubKey& base, const uint256& has
     return secp256k1_xonly_pubkey_tweak_add_check(secp256k1_context_verify, m_keydata.begin(), parity, &base_point, hash.begin());
 }
 
+bool CPubKey::TweakMulVerify(const CPubKey& res, const uint256& tweak) const
```

Comment:
I moderately prefer this method is deleted and its operation is inlined.  TweakMulVerify as written doesn't really handle non-compressed keys, but as a method of CPubKey it would need to.

Reply:
Agreed

Reply:
On second thought,

- There is also derive method on `CPubKey` that does not make sense for Uncompressed keys. I can add an assert to ensure it. 
- The current code architecture makes it cleaner to use all libsecp methods inside the pubkey module. This code would be the only code in `interpreter.cpp` that would require context objects, and directly accesses the libsecp methods. 
- Agreed with naming. I can interchange the position of `res` and `this` in the method.

Reply:
Fairpoint about the encapsulation of secp256k1.